### PR TITLE
Add CI

### PR DESCRIPTION
**Pull request by zultron**
Add `industrial_ci` configuration based on MoveIt's.  Tests partly fail; this is expected.

### CODE REVIEW

**Comment by zultron** [comment]
@vatanaksoytezer

**Comment by vatanaksoytezer** [comment]
@zultron you seem to have a different failure in GHA though: `rospkg.common.ResourceNotFound: hal_402_device_mgr`. I didn't run ici locally so I can't say much about ici itself. I will take a look into what the errors are, soon.

Also there is some unneded stuff here that I am seeing (that is probably copied over moveit repositories). For example:

1) Why do you need all the docker actions here?
2) Why do you need prerelease?
3) Why do you need moveit packages and moveit dockers for CI?

etc.

So you would typically only need pre-commit action and pre-commit configs (if you want formatting) and a very plain ici job like https://github.com/ros-planning/random_numbers/blob/master/.github/workflows/industrial_ci_action.yaml. The reason moveit is built on docker is just to save time on dependency installation (ie: docker only contains necessary dependencies), I don't think you need that for a single package.

**Comment by zultron** [comment]
> @zultron you seem to have a different failure in GHA though: `rospkg.common.ResourceNotFound: hal_402_device_mgr`. I didn't run ici locally so I can't say much about ici itself. I will take a look into what the errors are, soon.

Fixed that.  Now the failures are just the expected unit test failures.  No need to worry about those, since they'll be taken care of during the refactor.

> Also there is some unneded stuff here that I am seeing (that is probably copied over moveit repositories). For example:
> 
>     1. Why do you need all the docker actions here?
> 
>     2. Why do you need prerelease?
> 
>     3. Why do you need moveit packages and moveit dockers for CI?
> 
> 
> etc.
> 
> So you would typically only need pre-commit action and pre-commit configs (if you want formatting) and a very plain ici job like https://github.com/ros-planning/random_numbers/blob/master/.github/workflows/industrial_ci_action.yaml. The reason moveit is built on docker is just to save time on dependency installation (ie: docker only contains necessary dependencies), I don't think you need that for a single package.

You're right, I was following Tyler's advice and copied over the MoveIt CI config, then tweaked that.  I think I took care of the parts that would be hard for someone without much experience with the Machinekit codebase, i.e. the `UPSTREAM_WORKSPACE` configuration, and without visibility into the ZA6 project, mainly the formatter configs.

I'd be grateful if you wanted to step in and tighten up the rest of the configuration, or else add these critiques to another issue and I'll take care of them down the line when I have more time.  Thanks, Vatan!

**Comment by vatanaksoytezer** [comment]
> I'd be grateful if you wanted to step in and tighten up the rest of the configuration, or else add these critiques to another issue and I'll take care of them down the line when I have more time. Thanks, Vatan!

Sure I don't mind stepping in. It might take a few days before I can spare some time on this though if you don't mind.

**Comment by zultron** [comment]
Although tests aren't passing and @vatanaksoytezer wants to improve the CI config, I'd still like to merge this more or less as-is to unblock following PRs that are stacking up.

This code is getting a refactor and tests are being redone anyway, so it's not worth the effort to make tests pass.  I can simply disable them if the failing status is a problem.

**Comment by zultron** [comment]
@MarqRazz you should have an invite to collaborate on this repo; let me know if not.

File: .github/workflows/ci.yaml
Comment (vatanaksoytezer) [inline .github/workflows/ci.yaml]: You can remove build all on branches, since it will still build on all PRs

File: .github/workflows/docker.yaml
Comment (vatanaksoytezer) [inline .github/workflows/docker.yaml]: Remove the docker file since it is not relevant.

File: .github/workflows/prerelease.yaml
Comment (vatanaksoytezer) [inline .github/workflows/prerelease.yaml]: Remove the pre-release test since this is not relevant.

**Review by vatanaksoytezer — APPROVED** [review]
So added a few comments to cleanup a little. If this is blocking other PRs I will say go ahead and merge. I can deal with the further cleaning later on.

**Review by MarqRazz — COMMENTED** [review]
@zultron just so you know in the future you should try to separate out code changes from pre-commit/addition of CI. It makes it hard to understand where you are modifying the framework or formatting was just applied.

Is the change you are attempting to make just a rename (I see that's the name of the PR) or do you have other things you are also applying here?

**Comment by zultron** [comment]
> @zultron just so you know in the future you should try to separate out code changes from pre-commit/addition of CI. It makes it hard to understand where you are modifying the framework or formatting was just applied.
> 
> Is the change you are attempting to make just a rename (I see that's the name of the PR) or do you have other things you are also applying here?

@MarqRazz You're absolutely right.  I submitted #2 , isolating just the renaming changes so you can see exactly what went into it.

This PR now depends on that, so it's marked as "draft" until that one has been merged.  Name & description updated.

File: .github/workflows/ci.yaml
Comment (zultron) [reply .github/workflows/ci.yaml]: Done.

File: .github/workflows/docker.yaml
Comment (zultron) [reply .github/workflows/docker.yaml]: Done.

File: .github/workflows/prerelease.yaml
Comment (zultron) [reply .github/workflows/prerelease.yaml]: Done.

**Review by zultron — COMMENTED** [review]
@vatanaksoytezer Thanks for the suggestions.  I've added them all.

In re. "remove build all on branches" in `ci.yaml`, if I want to check if my private branch will pass tests before submitting a PR, is there a more straightforward way than re-adding that change, checking the GH Actions result, and removing it before submitting the PR?

**Comment by zultron** [comment]
Anyone else want look at this, or can we merge?  Thanks!

**Comment by dchaniel** [comment]
@zultron the nuances of CI are pretty out of my comfort zone so I'll trust @MarqRazz  and @vatanaksoytezer  here, I think good to merge if they're good with it

File: .github/workflows/format.yaml
Comment (vatanaksoytezer) [inline .github/workflows/format.yaml:9]: @zultron can you add make it so it will only run on main push.

**Comment by zultron** [comment]
I didn't actually mean to merge this.  ;(  I'll set up a new PR to address @vatanaksoytezer 's suggestions.

File: .github/workflows/format.yaml
Comment (zultron) [reply .github/workflows/format.yaml:9]: Sorry @vatanaksoytezer , can you look at #3 for this update?

Comment (zultron) [reply .github/workflows/format.yaml:9]: I put in a commit addressing this before accidentally merging.  Can you check out 04959cf7fcf87473eb4b6b096f61ec8120072829 to confirm it's what you expected?

**Pull request merged**
